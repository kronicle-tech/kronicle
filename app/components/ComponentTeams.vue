<template>
  <ul v-if="componentTeams && componentTeams.length > 0">
    <li
      v-for="(componentTeam, componentTeamIndex) in componentTeams"
      :key="componentTeamIndex"
    >
      <TeamName :team="{ id: componentTeam.teamId }" />
      <b-badge v-if="componentTeam.type === 'primary'" variant="info">
        Primary Team
      </b-badge>
      <b-badge v-else-if="componentTeam.type === 'previous'" variant="light">
        Previous Team
      </b-badge>
      <Markdown
        v-if="componentTeam.description"
        :markdown="componentTeam.description"
      />
    </li>
  </ul>
</template>

<style>
.team-name + .badge {
  margin-left: 0.6em;
}
</style>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { BBadge } from 'bootstrap-vue'
import { ComponentTeam } from '~/types/kronicle-service'
import Markdown from '~/components/Markdown.vue'
import TeamName from '~/components/TeamName.vue'

export default Vue.extend({
  components: {
    'b-badge': BBadge,
    Markdown,
    TeamName,
  },
  props: {
    componentTeams: {
      type: Array as PropType<ComponentTeam[]>,
      default: undefined,
    },
  },
})
</script>
