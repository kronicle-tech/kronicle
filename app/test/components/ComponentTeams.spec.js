import { mount } from '@vue/test-utils'
import ComponentTeams from '~/components/ComponentTeams.vue'

describe('Teams', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(ComponentTeams, { propsData })
  }

  beforeEach(() => {
    propsData = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when teams prop is not set', () => {
    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
    })
  })

  describe('when teams prop is set to an empty array', () => {
    beforeEach(() => {
      propsData.teams = []
    })

    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
    })
  })

  describe('when teams prop is set to one team', () => {
    beforeEach(() => {
      propsData.teams = [
        {
          id: 'test-team-1',
        },
      ]
    })

    test('renders an unordered list showing the team', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when teams prop is set to a team with type of primary', () => {
    beforeEach(() => {
      propsData.teams = [
        {
          id: 'test-team-1',
          type: 'primary',
        },
      ]
    })

    test('renders the team with a primary team badge', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when teams prop is set to a team with type of previous', () => {
    beforeEach(() => {
      propsData.teams = [
        {
          id: 'test-team-1',
          type: 'previous',
        },
      ]
    })

    test('renders the team with a previous team badge', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when teams prop is set to a team with a description', () => {
    beforeEach(() => {
      propsData.teams = [
        {
          id: 'test-team-1',
          description: 'Test Description 1',
        },
      ]
    })

    test('renders the team with a description', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when teams prop is set to a team with type of primary and a description', () => {
    beforeEach(() => {
      propsData.teams = [
        {
          id: 'test-team-1',
          type: 'primary',
          description: 'Test Description 1',
        },
      ]
    })

    test('renders the team with a primary team badge and a description', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when teams prop is set to one team with Markdown in the description', () => {
    beforeEach(() => {
      propsData.teams = [
        {
          id: 'test-team-1',
          description: 'Text with *bold* formatting',
        },
      ]
    })

    test('renders the team with the Markdown converted to HTML', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when teams prop is set to multiple teams', () => {
    beforeEach(() => {
      propsData.teams = [
        {
          id: 'test-team-1',
        },
        {
          id: 'test-team-2',
        },
        {
          id: 'test-team-3',
        },
      ]
    })

    test('renders an unordered list showing the teams', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
