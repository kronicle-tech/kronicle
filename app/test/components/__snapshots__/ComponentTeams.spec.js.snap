// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams when teams prop is set to a team with a description renders the team with a description 1`] = `""`;

exports[`Teams when teams prop is set to a team with type of previous renders the team with a previous team badge 1`] = `""`;

exports[`Teams when teams prop is set to a team with type of primary and a description renders the team with a primary team badge and a description 1`] = `""`;

exports[`Teams when teams prop is set to a team with type of primary renders the team with a primary team badge 1`] = `""`;

exports[`Teams when teams prop is set to multiple teams renders an unordered list showing the teams 1`] = `""`;

exports[`Teams when teams prop is set to one team renders an unordered list showing the team 1`] = `""`;

exports[`Teams when teams prop is set to one team with Markdown in the description renders the team with the Markdown converted to HTML 1`] = `""`;
